neuralforecast/data/datasets/favorita.py

```diff
@@ -0,0 +1,943 @@
+# AUTOGENERATED! DO NOT EDIT! File to edit: nbs/data_datasets__favorita.ipynb (unless otherwise specified).
+
+__all__ = ['logger', 'CodeTimer', 'check_nans', 'check_items', 'one_hot_encoding', 'nonzero_indexes_by_row',
+           'numpy_balance', 'numpy_ffill', 'numpy_bfill', 'distance_to_holiday', 'make_holidays_distance_df',
+           'Favorita']
+
+# Cell
+import shutil
+from py7zr import unpack_7zarchive
+# shutil.register_unpack_format('7zip', ['.7z'], unpack_7zarchive)
+
+# import mkl
+# mkl.set_num_threads(56)
+
+# Cell
+import os
+import gc
+import sys
+import time
+import tqdm
+import timeit
+
+import numpy as np
+import pandas as pd
+from datetime import datetime
+from datetime import timedelta
+
+import sklearn.preprocessing as preprocessing
+from sklearn.preprocessing import OneHotEncoder
+from sklearn.preprocessing import StandardScaler
+
+# Cell
+import logging
+import requests
+import subprocess
+import zipfile
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple, Union
+
+from dataclasses import dataclass
+from .utils import (
+    download_file, Info, TimeSeriesDataclass
+)
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+# Cell
+class CodeTimer:
```

Comment:
I think the `CodeTimer` class should be part of a general module, such as `neuralforecast.data.utils`.